tempodb/encoding/vparquet/schema.go

```diff
@@ -193,30 +198,29 @@ func attrToParquet(a *v1.KeyValue) Attribute {
 		_ = jsonMarshaler.Marshal(jsonBytes, a.Value) // deliberately marshalling a.Value because of AnyValue logic
 		p.ValueKVList = jsonBytes.String()
 	}
-	return p
 }
 
-func traceToParquet(id common.ID, tr *tempopb.Trace) Trace {
-
-	ot := Trace{
-		TraceIDText: util.TraceIDToHexString(id),
-		TraceID:     util.PadTraceIDTo16Bytes(id),
+func traceToParquet(id common.ID, tr *tempopb.Trace, ot *Trace) *Trace {
```

Comment:
This is a hard method to read currently, but I see why its necessary.  The `ot` signature extension looks like it is only used during `CreateBlock()` correct?  Could the calls to `extendReuseSlice` be hoisted into the `CreaetBlock()` method to avoid the extension?

Reply:
unsure how we would do that, but if you have a suggestion please lay it on me. an improvement on this code that I don't think would be too difficult would be to pool the slices. currently we are relying on this one trace object which we repeatedly overwrite to "hold" our buffers.

i'm currently working on a "leveled" pooling PR for the upstream parquet library. I could likely do something similar here which would make this an easier read and more effectively use memory.

Reply:
Leveled pool added and used.